internal/controller/article.go

```diff
@@ -0,0 +1,80 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/newUser1337/task-news/internal/entity"
+
+	"github.com/gorilla/mux"
+)
+
+type articleUsecase interface {
+	GetArticles(context.Context) ([]*entity.NewsletterNewsItemResponse, error)
+	GetArticle(context.Context, int64) (*entity.NewsArticleResponse, error)
+}
+
+type ArticleHandler struct {
+	article articleUsecase
+}
+
+func NewArticlesHandler(article articleUsecase) *ArticleHandler {
+	return &ArticleHandler{
+		article: article,
+	}
+}
+
+func (a *ArticleHandler) GetArticles(w http.ResponseWriter, req *http.Request) {
+	articles, err := a.article.GetArticles(req.Context())
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Println("failed to get articles", err)
+		return
+	}
+	response, err := json.Marshal(articles)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Println("failed to marshal articles", err)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	if _, err := w.Write(response); err != nil {
+		log.Printf("GetArticles: failed to write response %s\n", err)
+	}
+}
```

Comment:
ideally you're supposed to have helper functions for returning successful and error response instead of copying this code from one handler to another

Reply:
fixed